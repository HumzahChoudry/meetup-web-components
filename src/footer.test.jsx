import React from 'react';
import { shallow, mount } from 'enzyme';

import {
	Footer,
	FooterCategory,
	LanguageSelectInput,
	SocialIconsList,
	languagesAsOptions
} from './Footer';

const APP_BADGES_CLASSNAME = 'appBadges';
const DEFAULT_LANG = 'en-US';
const FRENCH_LANG = 'fr-FR';
const onLanguageSelect = jest.fn();

const footerLinkSets = [
	{
		header: 'Your Account',
		items: [
			<a href='/register'>Settings</a>,
			<a href="/logout">Log out</a>,
			<a href="/help">Help</a>
		]
	},
	{
		header: 'Discover',
		items: [
			<a href='/groups'>Groups</a>,
			<a href="/calendar">Calendar</a>,
			<a href="/topics">Topics</a>,
			<a href="/cities">Cities</a>
		]
	},
	{
		header: 'Meetup',
		items: [
			<a href='/about'>About</a>,
			<a href="/pro">Meetup Pro</a>,
			<a href="/jobs">Careers</a>,
			<a href="/apps">Apps</a>,
			<a href="/meetup_api">API</a>
		]
	},
];

const subFooterLinks = [
	<a href="/terms">Terms</a>,
	<a href="/privacy">Privacy</a>,
	<a href="/cookie_policy">Cookies</a>,
];

const MOCK_DEFAULT_STATE = {
	localeCode: 'en-US',
	linkSets: footerLinkSets,
	onLanguageSelect: onLanguageSelect,
	subFooterLinks: subFooterLinks,
	appBadges: <div className={APP_BADGES_CLASSNAME} />,
	createMeetup: {
		text: 'Create a Meetup',
		link: '/create/'
	},
	media: { isAtLargeUp: false },
};

const MOCK_AUTHENTICATED_STATE = { isLoggedIn: true };

describe('Footer', () => {
	const footerComponent = shallow(<Footer {...MOCK_DEFAULT_STATE} />);
	it('exists', () => {
		expect(footerComponent).toMatchSnapshot();
	});
	it('should render social media icon list', () => {
		const socialMediaIconList = footerComponent.find(SocialIconsList);
		expect(socialMediaIconList.exists()).toBe(true);
	});
	it('should render FooterCategory with linkSet', () => {
		const footerCategories = shallow(<FooterCategory {...footerLinkSets[0]} />);
		expect(footerCategories).toMatchSnapshot();
	});
});

describe('Footer unauthenticated user', () => {
	const footerComponent = shallow(<Footer {...MOCK_DEFAULT_STATE} />);
	it('shows language picker', () => {
		const langSelect = footerComponent.find(LanguageSelectInput);

		expect(langSelect.length).toBeGreaterThan(0);
	});
});

describe('Footer authenticated user', () => {
	const footerComponent = shallow(
		<Footer isLoggedIn {...MOCK_DEFAULT_STATE} {...MOCK_AUTHENTICATED_STATE} />
	);
	it('does *not* show language picker', () => {
		const langSelect = footerComponent.find(LanguageSelectInput);
		expect(langSelect.length).toBe(0);
	});
});

describe('LanguageSelectInput', () => {
	it('exists', () => {
		const langSelectInput = shallow(<LanguageSelectInput language={DEFAULT_LANG} />);
		expect(langSelectInput).toMatchSnapshot();
	});
	it('should render the correct language options', () => {
		const langSelectInput = shallow(<LanguageSelectInput language={DEFAULT_LANG} />);
		expect(langSelectInput.prop('options')).toEqual(languagesAsOptions);
	});
	it('should render the correct language value', () => {
		const langSelectInput = mount(<LanguageSelectInput language={DEFAULT_LANG} />);
		const selectWrapper = langSelectInput.find('select');
		expect(selectWrapper.prop('value')).toEqual(DEFAULT_LANG);
	});
	it('should call onLanguageSelect', () => {
		const footerWrapper = mount(<Footer {...MOCK_DEFAULT_STATE} />);
		const selectWrapper = footerWrapper.find(LanguageSelectInput).find('select');
		expect(onLanguageSelect).not.toHaveBeenCalled();
		selectWrapper.simulate('change');
		expect(onLanguageSelect).toHaveBeenCalled();
	});
});

describe('SocialIconsList', () => {
	it('exists', () => {
		const socialIconList = shallow(<SocialIconsList language={FRENCH_LANG} />);
		expect(socialIconList).toMatchSnapshot();
	});
});

