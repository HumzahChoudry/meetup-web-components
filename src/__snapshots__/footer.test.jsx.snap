// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer exists 1`] = `
<footer
  className="column-item column-item--shrink"
  id="mupFooter"
>
  <Component
    className="footerStripe-main"
    inverted={true}
  >
    <Component
      className="bounds--footer"
      narrow={true}
    >
      <Component
        hasSeparator={true}
      >
        <Component>
          <a
            className="link link--white"
            href="/create/"
          >
            Create a Meetup
          </a>
        </Component>
      </Component>
      <Component
        className="border--none"
        hasSeparator={true}
      >
        <Component
          direction="column"
          switchDirection="large"
        >
          <Component
            className="margin--bottom"
          >
            <Component
              direction="column"
              switchDirection="medium"
            >
              <Component
                key="linkSet-0"
              >
                <FooterCategory
                  header="Your Account"
                  items={
                    Array [
                      <a
                        href="/register"
                    >
                        Settings
                    </a>,
                      <a
                        href="/logout"
                    >
                        Log out
                    </a>,
                      <a
                        href="/help"
                    >
                        Help
                    </a>,
                    ]
                  }
                />
              </Component>
              <Component
                key="linkSet-1"
              >
                <FooterCategory
                  header="Discover"
                  items={
                    Array [
                      <a
                        href="/groups"
                    >
                        Groups
                    </a>,
                      <a
                        href="/calendar"
                    >
                        Calendar
                    </a>,
                      <a
                        href="/topics"
                    >
                        Topics
                    </a>,
                      <a
                        href="/cities"
                    >
                        Cities
                    </a>,
                    ]
                  }
                />
              </Component>
              <Component
                key="linkSet-2"
              >
                <FooterCategory
                  header="Meetup"
                  items={
                    Array [
                      <a
                        href="/about"
                    >
                        About
                    </a>,
                      <a
                        href="/pro"
                    >
                        Meetup Pro
                    </a>,
                      <a
                        href="/jobs"
                    >
                        Careers
                    </a>,
                      <a
                        href="/apps"
                    >
                        Apps
                    </a>,
                      <a
                        href="/meetup_api"
                    >
                        API
                    </a>,
                    ]
                  }
                />
              </Component>
            </Component>
          </Component>
          <Component
            shrink={true}
          >
            <Component
              direction="column"
              noGutters={true}
            >
              <Component
                shrink={true}
              >
                <Component
                  className="align--center atMedium_align--left"
                >
                  <SocialIconsList
                    localeCode="en-US"
                  />
                </Component>
              </Component>
              <Component>
                <Component
                  direction="column"
                  justify={null}
                >
                  <Component
                    className="margin--top"
                    shrink={true}
                  >
                    <Component>
                      <LanguageSelectInput
                        currentLocaleCode="en-US"
                        onChange={[Function]}
                      />
                    </Component>
                  </Component>
                  <Component
                    shrink={true}
                  >
                    <Component
                      className="align--center atMedium_align--left margin--top"
                    >
                      <AppBadges
                        getAppStorePhoto={[Function]}
                        language="en"
                      />
                    </Component>
                  </Component>
                </Component>
              </Component>
            </Component>
          </Component>
        </Component>
      </Component>
    </Component>
  </Component>
  <Component
    className="footerStripe-legal border--none"
    inverted={true}
  >
    <Component>
      <div
        className="padding--all"
      >
        <Component
          className="text--small align--center atMedium_align--left"
          items={
            Array [
              <span>
                © Meetup 2018
            </span>,
              <a
                href="/terms"
            >
                Terms
            </a>,
              <a
                href="/privacy"
            >
                Privacy
            </a>,
              <a
                href="/cookie_policy"
            >
                Cookies
            </a>,
            ]
          }
          separator="·"
        />
      </div>
    </Component>
  </Component>
</footer>
`;

exports[`Footer should render FooterCategory with linkSet 1`] = `
<Component>
  <h4
    className="text--bold margin--bottom"
  >
    Your Account
  </h4>
  <Component
    className="text--secondary text--small"
    columns={
      Object {
        "default": 2,
        "medium": 1,
      }
    }
    itemClassNames="footer-item"
    items={
      Array [
        <a
          href="/register"
      >
          Settings
      </a>,
        <a
          href="/logout"
      >
          Log out
      </a>,
        <a
          href="/help"
      >
          Help
      </a>,
      ]
    }
  />
</Component>
`;

exports[`LanguageSelectInput exists 1`] = `
<WithErrorList(SelectInput)
  label="Language"
  labelClassName="visibility--a11yHide"
  name="languagePicker"
  options={
    Array [
      Object {
        "label": "English",
        "value": "en-US",
      },
      Object {
        "label": "Deutsch",
        "value": "de-DE",
      },
      Object {
        "label": "English (Australia)",
        "value": "en-AU",
      },
      Object {
        "label": "Español",
        "value": "es",
      },
      Object {
        "label": "Español (España)",
        "value": "es-ES",
      },
      Object {
        "label": "Français",
        "value": "fr-FR",
      },
      Object {
        "label": "Italiano",
        "value": "it-IT",
      },
      Object {
        "label": "日本語",
        "value": "ja-JP",
      },
      Object {
        "label": "한국어",
        "value": "ko-KR",
      },
      Object {
        "label": "Nederlands",
        "value": "nl-NL",
      },
      Object {
        "label": "Português",
        "value": "pt-BR",
      },
      Object {
        "label": "Polski",
        "value": "pl-PL",
      },
      Object {
        "label": "Русский",
        "value": "ru-RU",
      },
      Object {
        "label": "ไทย",
        "value": "th-TH",
      },
      Object {
        "label": "Türkçe",
        "value": "tr-TR",
      },
    ]
  }
/>
`;

exports[`SocialIconsList renders for de-DE locale 1`] = `
<Component
  items={
    Array [
      <a
        href="https://www.facebook.com/meetupDE/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/MeetupDE/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for en-AU locale 1`] = `
<Component
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/Meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for en-US locale 1`] = `
<Component
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/Meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for es locale 1`] = `
<Component
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/MeetupES/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for es-ES locale 1`] = `
<Component
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/MeetupES/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for fr-FR locale 1`] = `
<Component
  items={
    Array [
      <a
        href="https://www.facebook.com/MeetupFR/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/MeetupFR/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for it-IT locale 1`] = `
<Component
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/MeetupIT/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for ja-JP locale 1`] = `
<Component
  items={
    Array [
      <a
        href="https://www.facebook.com/meetupjp/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/MeetupJP/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for ko-KR locale 1`] = `
<Component
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/Meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for nl-NL locale 1`] = `
<Component
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/Meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for pl-PL locale 1`] = `
<Component
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/Meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for pt-BR locale 1`] = `
<Component
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/MeetupBR/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for ru-RU locale 1`] = `
<Component
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/Meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for th-TH locale 1`] = `
<Component
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/Meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for tr-TR locale 1`] = `
<Component
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/Meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;
